### ISSUE TITLE
Training Bugs

### ISSUE THREAD

**Issue by Swastik166** [m-gallegos/SchNet4AIM#3]
Hey! Great work. 
I was trying to reproduce the training protocol for the models you provided. While doing so, I ran into two blocking problems when using SchNet4AIM with a JSON database and 2-particle properties:

In[ data](https://github.com/m-gallegos/SchNet4AIM/tree/main/src/SchNet4AIM/data)/atoms.py
In AtomsData, __len__ is defined inside the elif (s4aim.DBmode == "json"): branch, but there is no matching __getitem__ (only the DB-mode version exists). This breaks indexing/DataLoader usage for JSON datasets.

In [atomistic](https://github.com/m-gallegos/SchNet4AIM/tree/main/src/SchNet4AIM/atomistic)/model.py
In AtomisticModel.forward for s4aim.Pmode == "2p", I believe the logic for creating pair_rep is built with hard-coded batch=0 indexing and shapes that imply B=1. This causes incorrect behaviour or errors for B>1.

Could you please have a look at these?
Thank you

**Comment by ostreaterbl** [comment]
@Swastik166 hello. Did you manage to start the training process? I can't do anything.

**Comment by Swastik166** [comment]
Hi @ostreaterbl, yes, I can train now. The critical changes are to have __getitem__() method for json mode in atoms.py and change result[prop] to result[prop].squeeze() in line 39 of loss.py.
For 2p mode, keep the batch_size =1 only.

Once these changes are made, you can create a training script like the one in Schnetpack, and it should work.

**Comment by ostreaterbl** [comment]
Dear @Swastik166 , thanks for your answer. could you please show me your geiitem(), I would be very grateful
